### ISSUE TITLE
Windows Ruby 2.5.1 Test Failures

### ISSUE THREAD

**Issue by SilasReinagel** [yegor256/0pdd#188]
`rake` does not succeed on Windows 10 with Ruby 2.5.1 due to Test Failures. 

These two tests should be fixed.

```
========================================================================================================================D:/srcctrl/0pdd-dev/test/test_git_repo.rb:181:in `test_fetch_config'
     178:   end
     179:
     180:   def test_fetch_config
  => 181:     Dir.mktmpdir 'test' do |d|
     182:       repo = GitRepo.new(name: 'yegor256/0pdd', dir: d, uri: git(d))
     183:       repo.push
     184:       assert(repo.config['foo'])
D:/Languages/Ruby/Ruby25-x64/lib/ruby/2.5.0/tmpdir.rb:95:in `mktmpdir'
D:/Languages/Ruby/Ruby25-x64/lib/ruby/2.5.0/fileutils.rb:689:in `remove_entry'
D:/Languages/Ruby/Ruby25-x64/lib/ruby/2.5.0/fileutils.rb:1386:in `postorder_traverse'
D:/Languages/Ruby/Ruby25-x64/lib/ruby/2.5.0/fileutils.rb:1386:in `ensure in postorder_traverse'
D:/Languages/Ruby/Ruby25-x64/lib/ruby/2.5.0/fileutils.rb:691:in `block in remove_entry'
D:/Languages/Ruby/Ruby25-x64/lib/ruby/2.5.0/fileutils.rb:1329:in `remove'
D:/Languages/Ruby/Ruby25-x64/lib/ruby/2.5.0/fileutils.rb:1336:in `remove_dir1'
D:/Languages/Ruby/Ruby25-x64/lib/ruby/2.5.0/fileutils.rb:1351:in `platform_support'
D:/Languages/Ruby/Ruby25-x64/lib/ruby/2.5.0/fileutils.rb:1337:in `block in remove_dir1'
D:/Languages/Ruby/Ruby25-x64/lib/ruby/2.5.0/fileutils.rb:1337:in `rmdir'
Error: test_fetch_config(TestGitRepo): Errno::ENOTEMPTY: Directory not empty @ dir_s_rmdir - C:/Users/SILASR~1/AppData/Local/Temp/test20180402-462444-actjr5
========================================================================================================================...........E
========================================================================================================================D:/srcctrl/0pdd-dev/test/test_job_detached.rb:37:in `test_simple_scenario'
     34:     end
     35:     JobDetached.new(
     36:       FakeRepo.new, job
  => 37:     ).proceed
     38:   end
     39: end
D:/srcctrl/0pdd-dev/objects/job_detached.rb:35:in `proceed'
D:/srcctrl/0pdd-dev/objects/job_detached.rb:52:in `exclusive'
D:/srcctrl/0pdd-dev/objects/job_detached.rb:52:in `delete'
Error: test_simple_scenario(TestJobDetached): Errno::EACCES: Permission denied @ apply2files - C:/Users/SILASR~1/AppData/Local/Temp/0pdd-lock20180402-462444-qybegw
```

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) please, pay attention to this issue

**Comment by yegor256** [comment]
@SilasReinagel I added appveyor to check it

**Comment by 0crat** [comment]
Job [#188](https://github.com/yegor256/0pdd/issues/188) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
Bug was reported, see [§29](http://www.zerocracy.com/policy.html#29): +15 point(s) just awarded to @silasreinagel[/z](https://www.0crat.com/u/silasreinagel)

**Comment by 0crat** [comment]
The job [#188](https://github.com/yegor256/0pdd/issues/188) assigned to @kaze[/z](https://www.0crat.com/u/kaze), here is [why](https://www.0crat.com/footprint/C3T46CUJJ/1480751245); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html)com.zerocracy.Par@53ac0728

**Comment by 0crat** [comment]
@kaze[/z](https://www.0crat.com/u/kaze) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by 0crat** [comment]
The user @kaze[/z](https://www.0crat.com/u/kaze) resigned from [#188](https://github.com/yegor256/0pdd/issues/188), please stop working. Reason for job resignation: It is older than 10 days, see [§8](http://www.zerocracy.com/policy.html#8)

**Comment by 0crat** [comment]
Resigned on delay, see [§8](http://www.zerocracy.com/policy.html#8): -30 point(s) just awarded to @kaze[/z](https://www.0crat.com/u/kaze)

**Comment by 0crat** [comment]
The job [#188](https://github.com/yegor256/0pdd/issues/188) assigned to @golyalpha[/z](https://www.0crat.com/u/golyalpha), here is [why](https://www.0crat.com/footprint/C3T46CUJJ/1700908965); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be a monetary reward for this job

**Comment by golyalpha** [comment]
@SilasReinagel The code is directly from this repo?
What changes did you make?

**Comment by golyalpha** [comment]
@0crat Waiting on response from OP.

**Comment by SilasReinagel** [comment]
@golyalpha No changes. Just raw repo.

**Comment by 0crat** [comment]
> @0crat Waiting on response from OP. ([here](https://github.com/yegor256/0pdd/issues/188#issuecomment-398488886))

@golyalpha The impediment for [#188](https://github.com/yegor256/0pdd/issues/188) was registered successfully by @golyalpha[/z](https://www.0crat.com/u/golyalpha)

**Comment by golyalpha** [comment]
Because for me, rake fails on incorrect version of Ruby.
(got: 2.5.1, asks: 2.3.3)

**Comment by SilasReinagel** [comment]
@golyalpha The tests fail for me on Windows no matter which version of Ruby I am using 2.3.3 or 2.5.1

**Comment by golyalpha** [comment]
Was just asking to make sure I'm not working with the wrong version of the repo.

**Comment by golyalpha** [comment]
@SilasReinagel Can you give me steps to reproduce with a fresh install of Ruby 2.5.1?

**Comment by golyalpha** [comment]
@0crat Waiting for PR review (#222)

**Comment by 0crat** [comment]
> @0crat Waiting for PR review (#222) ([here](https://github.com/yegor256/0pdd/issues/188#issuecomment-399332639))

@golyalpha Job [#188](https://github.com/yegor256/0pdd/issues/188) is already on hold

**Comment by 0crat** [comment]
@rultor[/z](https://www.0crat.com/u/rultor) the issue is closed not by @SilasReinagel (its creator); I won't close the order; please, re-open it and ask @SilasReinagel to close it

**Comment by golyalpha** [comment]
@SilasReinagel could you please re-open and close the issue again? The problem should be fixed.

**Comment by SilasReinagel** [comment]
@golyalpha I don't have permissions.

@yegor256 Can you please re-open and close this issue? Tests no longer fail on Windows.

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) please, pay attention to this issue

**Comment by yegor256** [comment]
@SilasReinagel @golyalpha done

**Comment by 0crat** [comment]
There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 4@9f622663-8a45-40b0-97c3-f446fcf150da, thread: AsyncFlush-1
com.jcabi.xml.StrictXML[124] java.lang.IllegalArgumentException: 1 error(s) in XML document: -1:-1: Duplicate unique value [gh:yegor256/0pdd#188] declared for identity constraint of element "verbosity".
```

[0.23.1](https://github.com/zerocracy/farm): CID: [1760613143](https://www.0crat.com/C3T46CUJJ/1760613143), Type: "Close job", Author: "yegor256"

**Comment by 0crat** [comment]
Order was finished: +30 point(s) just awarded to @golyalpha[/z](https://www.0crat.com/u/golyalpha)

**Comment by 0crat** [comment]
The job [#188](https://github.com/yegor256/0pdd/issues/188) is now out of scope
